pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -570,6 +572,81 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param length the max length of the prefix strings for the string.
+   * @param regexChar the character for regex matching to be added to prefix strings generated. e.g. '^'
```

Comment:
What's the role of this?

Reply:
prefix can be ['a', 'ab', 'abc'] and it can also be ['^a', '^ab', '^abc']
suffix can be ['abc', 'bc', 'c'] and it can also be ['abc$', 'bc$', 'c$']
the regexChar is used for providing convenience for customer to match regex expressions

Reply:
In my opinion, it doesn't make sense to keep this in the OSS functions registry since this is too use-case specific. Can't the caller adjust their queries as required?

Reply:
hmm, the idea here is to support use cases that need to union the (prefixes, suffixes, ngram) column in one column, so the indexes size will reduced and easier to fit into memory.
I feel this is a relatively common use case to generate the prefix matcher and suffix matcher, so added the regex character as an optimal parameters. 

we can also do the following 
```
prefix(String input, int length)
prefixMatcher(String input, int length, String regexChar)

prefixes(String input, int minLength, int maxLength)
prefixMatchers(String input, int minLength, int maxLength, String regexChar)
```